Project-Aurora/Project-Aurora/Profiles/RocketLeague/GSI/GameState_RocketLeague.cs

```diff
@@ -1,43 +1,45 @@
 ﻿using Aurora.Profiles.RocketLeague.GSI.Nodes;
 using System;
+using System.ComponentModel;
 
 namespace Aurora.Profiles.RocketLeague.GSI
 {
+    public enum RLStatus
+    {
+        [Description("Menu")]
+        Undefined = -1,
+        Replay,
+        OnlineGame,
+        Freeplay,
+        Training,
+        Spectate
+    };
+
+    public class Game_RocketLeague : Node<Game_RocketLeague>
+    {
+        public RLStatus Status;
+
+        internal Game_RocketLeague(string json_data) : base(json_data)
+        {
+            Status = (RLStatus)GetInt("status");
+        }
+    }
+
     /// <summary>
     /// A class representing various information relating to Rocket League
     /// </summary>
     public class GameState_RocketLeague : GameState<GameState_RocketLeague>
     {
-        private Player_RocketLeague player;
-        private Match_RocketLeague match;
+        private Player_RocketLeague _Player;
+        private Match_RocketLeague _Match;
+        private Game_RocketLeague _Game;
 
-        /// <summary>
-        /// Information about the local player
-        /// </summary>
-        public Player_RocketLeague Player
-        {
-            get
-            {
-                if (player == null)
-                    player = new Player_RocketLeague("");
 
-                return player;
-            }
-        }
+        public Player_RocketLeague Player => _Player = new Player_RocketLeague(_ParsedData["player"]?.ToString() ?? "");
```

Comment:
This will _always_ assign `_Player` the value of a new node and return it, it will never use the cached version.

Consider using `_Player ?? (_Player = new Player_RocketLeague(_ParsedData["player"]?.ToString() ?? ""));` instead.